x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_initializer.tsx

```diff
@@ -6,64 +6,118 @@
 import type { SingleMetricViewerEmbeddableInput } from '..';
+import { ML_PAGES } from '../../../common/constants/locator';
 import { SeriesControls } from '../../application/timeseriesexplorer/components/series_controls';
 import {
   APP_STATE_ACTION,
   type TimeseriesexplorerActionType,
 } from '../../application/timeseriesexplorer/timeseriesexplorer_constants';
+import { useMlLink } from '../../application/contexts/kibana';
+import { JobSelectorControl } from '../../alerting/job_selector';
 import type { SingleMetricViewerEmbeddableUserInput, MlEntity } from '..';
+import { getDefaultSingleMetricViewerPanelTitle } from './get_default_panel_title';
 
 export interface SingleMetricViewerInitializerProps {
   bounds: TimeRangeBounds;
-  defaultTitle: string;
   initialInput?: Partial<SingleMetricViewerEmbeddableInput>;
-  job: MlJob;
-  onCreate: (props: Partial<SingleMetricViewerEmbeddableUserInput>) => void;
+  mlApiServices: MlApiServices;
+  onCreate: (props: SingleMetricViewerEmbeddableUserInput) => void;
   onCancel: () => void;
 }
 
 export const SingleMetricViewerInitializer: FC<SingleMetricViewerInitializerProps> = ({
-  defaultTitle,
   bounds,
   initialInput,
-  job,
   onCreate,
   onCancel,
+  mlApiServices,
 }) => {
-  const isNewJob = initialInput?.jobIds !== undefined && initialInput?.jobIds[0] !== job.job_id;
+  const isMounted = useMountedState();
+  const newJobUrl = useMlLink({ page: ML_PAGES.ANOMALY_DETECTION_CREATE_JOB });
+  const [jobIds, setJobIds] = useState(initialInput?.jobIds ?? []);
 
-  const [panelTitle, setPanelTitle] = useState<string>(defaultTitle);
+  const initialValuesRef = useRef({
+    titleManuallyChanged: !!initialInput?.title,
+    isNewJob:
+      initialInput?.jobIds !== undefined && jobIds.length && initialInput?.jobIds[0] !== jobIds[0],
+  });
+
+  const [job, setJob] = useState<MlJob | undefined>();
+  const [panelTitle, setPanelTitle] = useState<string>(initialInput?.title ?? '');
   const [functionDescription, setFunctionDescription] = useState<string | undefined>(
     initialInput?.functionDescription
   );
   // Reset detector index and entities if the job has changed
   const [selectedDetectorIndex, setSelectedDetectorIndex] = useState<number>(
-    !isNewJob && initialInput?.selectedDetectorIndex ? initialInput.selectedDetectorIndex : 0
+    !initialValuesRef.current.isNewJob && initialInput?.selectedDetectorIndex
+      ? initialInput.selectedDetectorIndex
+      : 0
   );
   const [selectedEntities, setSelectedEntities] = useState<MlEntity | undefined>(
-    !isNewJob && initialInput?.selectedEntities ? initialInput.selectedEntities : undefined
+    !initialValuesRef.current.isNewJob && initialInput?.selectedEntities
+      ? initialInput.selectedEntities
+      : undefined
   );
-
+  const [errorMessage, setErrorMessage] = useState<string | undefined>();
   const isPanelTitleValid = panelTitle.length > 0;
 
+  useEffect(
+    function setUpPanel() {
+      if (isMounted()) {
```

Comment:
`isMounted` is supposed to be used with async state updates

Reply:
Good catch - moved in 7e3b004a5255f450ce23964cfa2235a68126dcf5